### ISSUE TITLE
Question about KD Regularization in code

### ISSUE THREAD

**Issue by GengZ** [yuanli2333/Teacher-free-Knowledge-Distillation#3]
According to the [KD Regularization code](https://github.com/yuanli2333/Teacher-free-Knowledge-Distillation/blob/2a7db7e81a35a30b88e959f19716e08cbb9067f3/my_loss_function.py#L47), the  $D_{KL}$ term in the total loss is written as $D_{KL}(p^t_{\tau}, p)$. 

While in the paper, according to Eqn (9) and Eqn (5), the $D_{KL}$ term in loss is written as $D_{KL}(p^t_{\tau}, p_{\tau})$.  

Since the ${\tau}$ is not always 1, the above two functions are not the same.

Sort of confused about this.
Thanks.

**Comment by yuanli2333** [comment]
Yes, for CIFAR100, this function is a little different, we found that it works better if we set temperature ${\tau}=1$ for the output of the targeted model on CIFAR100. But for Tiny-ImageNet and ImageNet, temperature ${\tau}$ is higher (same as our paper). This is weird but works better in practice. 
The thinking behind our method is to change the teacher with random accuracy (output a uniform distribution) to a teacher with higher accuracy.  Actually,  you can design your teacher based on our findings. Welcome to discuss if you manually designed a better teacher!

**Comment by GengZ** [comment]
Thanks for explanation 👍 
Still playing around with it :)

Here's another trivial question:
The loss function for virtual teacher (the second method) is 
$L_{reg} = (1 - \alpha) * H(p, q) + \alpha * D_{KL}(p^d_{\tau}, p_{\tau})$
Since the self-entropy for the ${p^d}$ is constant throughout the training process, will the performance fluctuate a lot if given 
$L_{reg} = (1 - \alpha) * H(p, q) + \alpha * H(p^d_{\tau}, p_{\tau})$,
where a constant self-entropy is added?

**Comment by yuanli2333** [comment]
The constant has no influence on gradients, so the performance should not fluctuate.
